packages/framework/src/Component/Cron/CronTimeResolver.php

```diff
@@ -31,7 +31,7 @@ protected function isMatchWithTimeString($value, $timeString)
         $matches = null;
         foreach ($timeValues as $timeValue) {
             if ($timeValue === '*'
-                || $timeValue === (string)$value
+                || (int)$timeValue === $value
```

Comment:
This implementation seems to me a bit risky as it suddenly allows malformed input times to be "0"

```php
(int)'0' === 0; // true
(int)'00' === 0; // true
(int)'a' === 0; // true
(int)'000' === 0; // true
(int)'01' === 0; // false only
(int)'0a1' === 0; // true
(int)'0e1' === 0; // true
(int)'*/4' === 0; // true -- this case should be handled differently, but is caught by "0"
```

The string comparison was more reasonable and safe in my opinion...
```diff
  $timeValue === '*'
  || $timeValue === (string)$value
+ || $timeValue === str_pad((string) $value, 2, '0', STR_PAD_LEFT)
  || preg_match....
```

Reply:
Fair point, I've used the `str_pad`ding of the value so both strings are of equal length.